.github/workflows/run-tests.yml

```diff
@@ -45,8 +45,19 @@ jobs:
       - name: Checkout neofs-node repository
         uses: actions/checkout@v3
         with:
-          repository: ${{ github.repository }}
-          ref: ${{ github.event.inputs.neofs_node_ref || github.head_ref || github.ref_name }}
+          repository: >
+            ${{
+              github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name ||
+              github.repository
```

Comment:
Have you tried using the `pull/ID/head` trick to keep the same repository?

Reply:
> Have you tried using the `pull/ID/head` trick to keep the same repository?

Yes, I tried it.
It's the PRs test results:
https://github.com/vvarg229/neofs-node/pull/4
https://github.com/vvarg229/neofs-node/pull/5

It's the ID/head test result:
https://github.com/vvarg229/neofs-node/actions/runs/4960370107

It's the push to master test result:
https://github.com/vvarg229/neofs-node/actions/runs/4977427703